frontend/hub/api/utils.tsx

```diff
@@ -7,11 +7,55 @@ import {
   postHubRequest,
   putHubRequest,
 } from './request';
+import { useTranslation } from 'react-i18next';
+import { requestGet } from '../../common/crud/Data';
+import { AnsibleAnsibleRepositoryResponse as Repository } from '../api-schemas/generated/AnsibleAnsibleRepositoryResponse';
+import { AnsibleAnsibleDistributionResponse as Distribution } from '../api-schemas/generated/AnsibleAnsibleDistributionResponse';
 
 function getBaseAPIPath() {
   return process.env.HUB_API_PREFIX;
 }
 
+type Results = { data: { results: Repository[] | Distribution[] } };
+
+export function useRepositoryBasePath(name: string, pulp_href?: string): Promise<string> {
+  const { t } = useTranslation();
+
+  const repoRequest = pulp_href
+    ? Promise.resolve({ name, pulp_href } as Repository)
+    : requestGet<Results>(pulpAPI`/repositories/ansible/ansible/?name=${name}&limit=1`).then(
+        firstResult
+      );
+
+  const distroRequest = requestGet<Results>(
+    pulpAPI`/distributions/ansible/ansible/?name=${name}&limit=1`
+  ).then(firstResult);
+
+  return Promise.all(repoRequest, distroRequest).then(async ([repository, distribution]) => {
```

Comment:
This needs to be an array
```suggestion
  return Promise.all([repoRequest, distroRequest]).then(async ([repository, distribution]) => {
```